lib/spack/spack/util/path.py

```diff
@@ -60,6 +61,63 @@
 SPACK_PATH_PADDING_CHARS = '__spack_path_placeholder__'
 
 
+def is_path_url(path):
+    url_tuple = urlparse(path)
+    if not url_tuple.netloc and \
+            url_tuple.scheme and url_tuple.scheme != 'file':
```

Comment:
It looks like this would return `True` for `/path/to/a/thing`:

```
>>> from six.moves.urllib.parse import urlparse
>>> urlparse('/path/to/a/thing')
ParseResult(scheme='', netloc='', path='/path/to/a/thing', params='', query='', fragment='')
```

(`not` binds more closely than `and` so I read this as `if (not url_tuple.netloc) and url_tuple.scheme and url_tuple.scheme != 'file'`)

I think you can just check if the `scheme` is present:

```
def is_path_url(path):
  url_tuple = urlparse(path)
  return bool(url_tuple.scheme)
```

Reply:
This assumes that filesystem paths, when formatted as URLs, ought to retain `/` instead of replacing with `\` on Windows (which I think is what we should do)

Reply:
Unfortunately, `urlparse` returns the scheme as `'c'` for paths such as `C:\\<any string here` i.e.
`
>>> from six.moves.urllib.parse import urlparse
>>> urlparse("C:\\hello\\world")
ParseResult(scheme='c', netloc='', path='\\hello\\world', params='', query='', fragment='')
`
netloc will generally be empty for non-urls of any type, and then can catch file urls from that point.

As to the second comment, I had considered that, and initially decided against that, but on hearing it again I agree.

Reply:
Updated in new commit.

Reply:
OK I think as long as this returns `False` for a string like `"/path/to/file"` that is the most important thing (I suspected it did not, so I was looking for confirmation on that)

> Unfortunately, urlparse returns the scheme as 'c' for paths such as C:\\<any string here 

Based on https://stackoverflow.com/questions/7109143/what-characters-are-valid-in-a-url, `\` wouldn't appear in a URL (unless it was URL-encoded in which case it wouldn't appear as a `\`), so possibly

```
def is_path_url(path):
  if not isinstance(six.string_types, path):
    return False # or maybe this function should always expect a string
  if '\' in path:
    return False
  url_tuple = urlparse(path)
  return bool(url_tuple.scheme)
```

Reply:
I think it's fair to assume that this function should expect a string. But I think this looks good, I'll update with this solution.

Reply:
Addressed in new commit.